app/models/result.rb

```diff
@@ -175,6 +175,168 @@ def view_token_expired?
     !self.view_token_expiry.nil? && Time.current >= self.view_token_expiry
   end
 
+  # Generate a PDF report for this result.
+  # Currently only supports PDF submission file (all other submission files are skipped).
+  def generate_print_pdf
+    marks = self.mark_hash
+    extra_marks = self.extra_marks
+    total_mark = self.get_total_mark
+    overall_comment = self.overall_comment
+    submission = self.submission
+    grouping = submission.grouping
+    assignment = grouping.assignment
+
+    # Make folder for temporary files
+    workdir = "tmp/print/#{self.id}"
+    FileUtils.mkdir_p(workdir)
+
+    # Constants used for PDF generation
+    logo_width = 80
+    line_space = 12
+    annotation_size = 20
+
+    # Generate front page
+    Prawn::Document.generate("#{workdir}/front.pdf") do
+      # Add MarkUs logo
+      image Rails.root.join('app/assets/images/markus_logo_big.png'),
+            at: [bounds.width - logo_width, bounds.height],
+            width: logo_width
+
+      font_families.update(
+        'Open Sans' => {
+          normal: Rails.root.join('vendor/assets/stylesheets/fonts/OpenSansEmoji.ttf'),
+          bold: Rails.root.join('vendor/assets/stylesheets/fonts/OpenSans-Bold.ttf')
+        }
+      )
+      font 'Open Sans'
+
+      # Title
+      formatted_text [{
+        text: "#{assignment.short_identifier}: #{assignment.description}", size: 20, styles: [:bold]
+      }]
+      move_down line_space
+
+      # Group members
+      grouping.accepted_students.includes(:user).find_each do |student|
+        text "#{student.user_name} - #{student.first_name} #{student.last_name}"
+      end
+      move_down line_space
+
+      # Marks
+      assignment.ta_criteria.order(:position).find_each do |criterion|
+        if criterion.is_a? RubricCriterion
+          mark = marks.dig(criterion.id, :mark)
```

Comment:
`mark` needs to be defined outside this statement for criteria types other than rubric.

Reply:
Great catch, thanks @pretendWhale!